Functions/MeanAbsoluteDeviation.py

```diff
@@ -0,0 +1,51 @@
+# Algorithm
+# 1. Calculate Mean
+# 2. Sum the absolute values of all values minus the mean
+# 3. Divide the sum by the number of values
+
+# Primary Algoirthm
+# -----------------
+def mean_absolute_deviation(input):
+    # Local variables
+    mean = calculate_mean(input)
+    dividend = 0
+    diviser = calculate_length(input)
+
+    # Calculate the dividend
+    for x in input:
+        dividend += absolute_value(x - mean)
+
+    # Return the Mean Absolute Deviation
+    return (float(dividend) / diviser)
+        
+# Helper Functions
+# ----------------
+def calculate_mean(input): 
+    sum = 0
+    for x in input:
+        sum += x
+    return (sum / calculate_length(input))
+
+def absolute_value(input):
+    if input < 0:
+        return -input
+    else:
+        return input
+
+def calculate_length(input):
```

Comment:
I know len() function exists but not sure if we could use it because of assignment constraints... so to be safe I made my own

Reply:
I think using the built in length function is fine, I think he just wants us to not use built in math functions (like importing and using numpy wouldn’t be allowed).